### ISSUE TITLE
Issue compiling in Ubuntu Docker container

### ISSUE THREAD

**Issue by mattbodholdt** [openca/libpki#22]
Issue compiling in Docker container

See this automated build log:  [https://hub.docker.com/r/mattbodholdt/openca-ocspd/builds/byd5yjuczumcddx82j86eql](https://hub.docker.com/r/mattbodholdt/openca-ocspd/builds/byd5yjuczumcddx82j86eql)

Installs successfully using this commit: 320e80589648f5a554304995b666de3274576176

Failure is:

```
Making all in openssl
make[3]: Entering directory '/libpki-master/src/drivers/openssl'
/bin/bash ../../../libtool  --tag=CC   --mode=compile gcc  -I. -I../../../src/libpki -I../.. -I/usr/include     -D__LIB_BUILD__  -I/usr/include -g -O2 -fstack-check -maccumulate-outgoing-args -Werror -Wfatal-errors -Wunused-variable   -I/usr/include/libxml2   -DLINUX -g -O2 -fstack-check -maccumulate-outgoing-args -Werror -Wfatal-errors -MT libpki_token_openssl_la-openssl_hsm.lo -MD -MP -MF .deps/libpki_token_openssl_la-openssl_hsm.Tpo -c -o libpki_token_openssl_la-openssl_hsm.lo `test -f 'openssl_hsm.c' || echo './'`openssl_hsm.c

libtool: compile:  gcc -I. -I../../../src/libpki -I../.. -I/usr/include -D__LIB_BUILD__ -I/usr/include -g -O2 -fstack-check -maccumulate-outgoing-args -Werror -Wfatal-errors -Wunused-variable -I/usr/include/libxml2 -DLINUX -g -O2 -fstack-check -maccumulate-outgoing-args -Werror -Wfatal-errors -MT libpki_token_openssl_la-openssl_hsm.lo -MD -MP -MF .deps/libpki_token_openssl_la-openssl_hsm.Tpo -c openssl_hsm.c  -fPIC -DPIC -o .libs/libpki_token_openssl_la-openssl_hsm.o

[91mIn file included from ../../libpki/pki.h:322:0,
                 from openssl_hsm.c:4:
../../libpki/pki_keyparams.h:18:29: error: unknown type name 'PKI_EC_KEY_FORM'
                             PKI_EC_KEY_FORM   curveForm,
                             ^
compilation terminated due to -Wfatal-errors.
[0m
Makefile:504: recipe for target 'libpki_token_openssl_la-openssl_hsm.lo' failed
make[3]: Leaving directory '/libpki-master/src/drivers/openssl'`
```

**Comment by jpawlowski** [comment]
Facing the same issue with similar build under Debian Stretch.
See Travis build log:
https://travis-ci.com/docker-home-automation-stack/dhas-pki-ocspd/jobs/142443994

**Comment by skrubly** [comment]
Any movement on a fix for this in 0.9.0? That commit looks pretty old and before the .8.9 release if I'm interpreting it correctly.

**Comment by mattbodholdt** [comment]
Does seem to work now

**Comment by mattbodholdt** [comment]
I recant that statement.. still a problem.  Newer build log: https://hub.docker.com/r/mattbodholdt/openca-ocspd/builds/bdkgem62has9d2kcxgnqxzr/

**Comment by michaelkrieger** [comment]
Same problem here with Debian Buster

**Comment by opencrypto** [comment]
This should be fixed in the latest code and OpenSSL 1.1.1+ in Ubuntu22 (closing the issue for now as it seems an old issue).
